service/handlers_assertion.go

```diff
@@ -78,14 +81,33 @@ func ModelAssertionHandler(w http.ResponseWriter, r *http.Request) ErrorResponse
 	signedAssertion, err := datastore.Environ.KeypairDB.SignAssertion(asserts.ModelType, assertionHeaders, []byte(""), model.BrandID, keypair.KeyID, keypair.SealedKey)
 	if err != nil {
 		logMessage("MODEL", "signing-assertion", err.Error())
-		return ErrorResponse{false, "signing-assertion", "", err.Error(), http.StatusInternalServerError}
+		return ErrorResponse{false, "signing-assertion", "", err.Error(), http.StatusBadRequest}
 	}
 
-	// Return successful JSON response with the signed text
-	formatSignResponse(true, "", "", "", signedAssertion, w)
+	// Add the account assertion to the assertions list
+	fetchAssertionFromStore(&assertions, asserts.AccountType, []string{model.BrandID})
+
+	// Add the account-key assertion to the assertions list
+	fetchAssertionFromStore(&assertions, asserts.AccountKeyType, []string{keypair.KeyID})
+
+	// Add the model assertion after the account and account-key assertions
+	assertions = append(assertions, signedAssertion)
+
+	// Return successful response with the signed assertions
+	formatAssertionResponse(true, "", "", "", assertions, w)
 	return ErrorResponse{Success: true}
 }
 
+func fetchAssertionFromStore(assertions *[]asserts.Assertion, modelType *asserts.AssertionType, headers []string) error {
+	accountKeyAssertion, err := account.FetchAssertionFromStore(modelType, headers)
+	if err != nil {
```

Comment:
Why is this method returning an error type if it always takes nil value?

Reply:
Good point!